app/logic/invitation_texts.rb

```diff
@@ -0,0 +1,240 @@
+# frozen_string_literal: true
+
+class InvitationTexts
+  STREAK_SIZE = 3
+
+  class << self
+    def student_message(protocol, protocol_completion)
+      curidx = current_index(protocol_completion)
+      sms_pool = []
+
+      sms_pool += special_conditions(protocol_completion, curidx)
+      sms_pool += threshold_conditions(protocol, protocol_completion, curidx) if sms_pool.empty?
+      sms_pool += default_and_streak_conditions(protocol_completion, curidx) if sms_pool.empty?
+
+      sms_pool.sample
+    end
+
+    def mentor_message(_protocol, _protocol_completion)
+      # curidx = current_index(protocol_completion)
+      sms_pool = []
+
+      sms_pool.sample
+    end
+
+    def nth_response_pool(_curidx)
+      raise 'method nth_response_pool not implemented by subclass!'
+    end
+
+    def rewards_threshold_pool(_threshold)
+      raise 'method rewards_threshold_pool not implemented by subclass!'
+    end
+
+    def default_pool
+      raise 'method default_pool not implemented by subclass!'
+    end
+
+    def about_to_be_on_streak_pool
+      raise 'method about_to_be_on_streak_pool not implemented by subclass!'
+    end
+
+    def on_streak_pool
+      raise 'method on_streak_pool not implemented by subclass!'
+    end
+
+    def first_responses_missed_pool
+      raise 'method first_responses_missed_pool not implemented by subclass!'
+    end
+
+    def missed_last_pool
+      raise 'method missed_last_pool not implemented by subclass!'
+    end
+
+    def missed_more_than_one_pool
+      raise 'method missed_more_than_one_pool not implemented by subclass!'
+    end
+
+    def missed_everything_pool
+      raise 'method missed_everything_pool not implemented by subclass!'
+    end
+
+    def rejoined_after_missing_one_pool
+      raise 'method rejoined_after_missing_one_pool not implemented by subclass!'
+    end
+
+    def rejoined_after_missing_multiple_pool
+      raise 'method rejoined_after_missing_multiple_pool not implemented by subclass!'
+    end
+
+    def missed_after_streak_pool
+      raise 'method missed_after_streak_pool not implemented by subclass!'
+    end
+
+    private
+
+    def default_and_streak_conditions(protocol_completion, curidx)
+      sms_pool = []
```

Comment:
volgens mij had je hier een reden voor, maar je zou ook gewoon overal een return kunnen doen, dan hoef je niet steeds te checken op `sms_pool.empty?`

```ruby
# Streak about to be 3
return about_to_be_on_streak_pool if protocol_completion[curidx][:streak] == STREAK_SIZE

# On bonus streak (== on streak > 3)
return on_streak_pool if protocol_completion[curidx][:streak] > STREAK_SIZE

# Default messages
return default_pool
```

Reply:
Ja dat is een mooiere notatie, maar niet equivalent.

Ik heb het nu zo gedaan omdat ik dacht dat je als je bijvoorbeeld bepaalde smsjes niet meer wilt versturen, dat het dan makkelijker is voor ons later om dan die messages gewoon die messages in dat array in studentinvitationtexts weg te halen, zonder te hoeven duiken in de logica van wanneer wat precies wordt aangeroepen. Ook was mijn idee dat we in grote delen dezelfde logica voor student en mentorteksten zouden kunnen gaan gebruiken, en dat we als bijv een bepaalde categorie voor mentoren niet van toepassing is, die sms pool gewoon een leeg array te laten returnen, waardoor het programma dan automatisch door gaat naar de volgende test.